mayastor/src/core/thread.rs

```diff
@@ -23,13 +23,26 @@ pub enum Error {
     InvalidThread {},
 }
 
-#[derive(Debug, PartialEq, Clone, Copy)]
-/// struct that wraps an SPDK thread. The name thread is chosen poorly and
-/// should not be confused with an actual thread. Consider it more to be
-/// analogous to a container to which you can submit work and poll it to drive
+/// A wrapper around an SPDK thread.
+///
+/// The name thread is chosen poorly and should not be confused with an actual thread. Consider it
+/// more to be analogous to a container to which you can submit work and poll it to drive
 /// the submitted work to completion.
+///
+/// # Safety
+///
+/// SPDK considers this threadsafe, so it is marked Send/Sync.
+///
+/// See: https://github.com/spdk/spdk/blob/fb27c710f2c5ee4e666d2b366237e04c08dda977/include/spdk_internal/thread.h#L95-L136
+#[derive(Debug, PartialEq, Clone, Copy)]
 pub struct Mthread(NonNull<spdk_thread>);
 
+// Safety: SPDK considers the inner `spdk_thread` threadsafe.
+unsafe impl Send for Mthread {}
+
+// Safety: SPDK considers the inner `spdk_thread` threadsafe.
+unsafe impl Sync for Mthread {}
```

Comment:
I'm not about the sync part. Although in the future, the ability to move threads from/to cores is desired, nothing AFAIK, will prevent two cores/threads call spdk_thread_poll() on each. Having that said, I don't know if that is a problem right now?
As this code works without the sync marker, can, to be sure, omit it?

Reply:
It seems like they are sync though? I see what looks like lockless rings in https://github.com/openebs/spdk/blob/050ecb8c3a277e3a2dc8ccb1e370fccde642bf11/lib/thread/thread.c#L890-L944 and you can see in https://github.com/openebs/spdk/blob/050ecb8c3a277e3a2dc8ccb1e370fccde642bf11/lib/thread/thread.c#L951 there is certainly atomics happening.